### ISSUE TITLE
DRV8825 Solder and Motor Torque

### ISSUE THREAD

**Issue by elciomsantana** [Deimos1994/ESP32_Robot#1]
I was building a robot inspired by what you were inspired by, but with your code I managed to balance it, but I couldn't control it via WIFI, as it always falls, do you think the DAC soldier on the DRV8825 could be the problem?

Currently the DAC is only on the negative, as shown in the attached photo ...
![image](https://user-images.githubusercontent.com/76231121/102667457-31a17080-4168-11eb-9236-61613742e61e.png)

Another point, when removing the potentiometer and soldering it to the DAC, do I also need to solder this tiny wire to the CHIP SMD? How did you do it?
![image](https://user-images.githubusercontent.com/76231121/102667521-55fd4d00-4168-11eb-9c96-0d54fe1af980.png)

Thanks for the help, I developed a head with 1 ultrasonic sensor and 1 laser that turns 180 degrees horizontally and 120 degrees vertically, so it can move around alone.

I have esp32cam also that I intend to attach to that head in the future.

**Comment by Deimos1994** [comment]
@elciomsantana Hi, what do you mean by "Currently the DAC is only on the negative" ?
Did you connect everything according to the circuit? you need to remove the trimmer potentiometers on DVR8825 modules and connect a wire to the resistors (as shown on my circuit), without it the robot is not very stable.
No, you don't need to solder a tiny wire to the CHIP SMD, I did not see any difference with it or without it.

**Comment by elciomsantana** [comment]
https://user-images.githubusercontent.com/76231121/102816179-55ed8f00-43ac-11eb-9a8a-4f636d67fb22.mp4

![ae3622db-5803-4893-b64a-5bfcf54684bc (1)](https://user-images.githubusercontent.com/76231121/102816316-94834980-43ac-11eb-8319-9fdd3cf99f85.jpg)

**Comment by elciomsantana** [comment]
I removed the potentiometer and connected the DAC. Thanks!

Is there a problem connecting the DRV8825 drivers to the 5.5V current instead of the 3.3V ESP32 DEV kit itself?
I say in relation to the current calculation of the DAC because I am currently connecting everything in 5.5V.

Robot is standing still, I can rotate him very slowly by remote control, but if I put him to move forward or turn fast he falls.
If I push it, it gets out of control and falls, you know what it could be?

The diameter of the wheel is approximately 12cm, and the height of the MPU is 50cm, I do not know if that influence.

**Comment by Deimos1994** [comment]
@elciomsantana On the video it seems like the only one stepper motor is working. Check the second motor, make sure it works.
`Is there a problem connecting the DRV8825 drivers to the 5.5V current instead of the 3.3V ESP32 DEV kit itself?`
Well, I did not test it with 5V. Why do you even use 5.5V for DRV8825 logic? I just don't understand it. Use 3.3V like in the original circuit, DRV8825 works fine with 3.3V.

**Comment by elciomsantana** [comment]
Yes, in this video I was operating with 1 stepper motor only, but this has already been solved, I had burnt a DRV8825.

I turned on 5.5v because it was simpler to connect, and the MPU works better with voltage above 3.3v with longer wires.

I will do this modification, another thing that is different in my project, is the FAULT wire that I did not connect at all, and the SLP + RST jumper are connected on the 5v as shown in the image below

![image](https://user-images.githubusercontent.com/76231121/102998562-23f04000-4506-11eb-836b-b6e0237e0c04.png)

Two StepMotors Working now:

https://user-images.githubusercontent.com/76231121/102999021-0cfe1d80-4507-11eb-9238-99e63c5bd593.mp4

No potentiometer : 
![image](https://user-images.githubusercontent.com/76231121/102998833-b55fb200-4506-11eb-9edf-f76d08ef537b.png)

**Comment by Deimos1994** [comment]
@elciomsantana I think it should work with all your modification.
Did you tune PID controller values for your robot?

**Comment by elciomsantana** [comment]
I change this values but not success
![image](https://user-images.githubusercontent.com/76231121/103027856-68470480-4535-11eb-8fc2-8776c08f735a.png)


I did all the modifications but it is still unstable when I try to use the remote control or push it ...
The whole system is running with the ESP32 3.3v

![image](https://user-images.githubusercontent.com/76231121/103029464-9ed24e80-4538-11eb-8213-05976d8feab9.png)

![image](https://user-images.githubusercontent.com/76231121/103029571-d80abe80-4538-11eb-8087-6f6c7256fdcb.png)

**Comment by Deimos1994** [comment]
@elciomsantana `I change this values but not success`
1. Don't change that, set it as it was before.
2. Then go to http://192.168.4.1/
![Annotation 2020-12-23 194625](https://user-images.githubusercontent.com/31592485/103037789-58e7ab80-4575-11eb-990c-a018749af3b4.png)
Change PID tuning: Angle and Pos and make the robot stable
3. after that save your values in the scetch
```cpp
  pidAngle.setParameters(1.2,0.57,0.19,5.8);
  pidPos.setParameters(1.1,1.85,0.72,20);
  pidSpeed.setParameters(1.8,3.6,0.65,0.6);
```
4. Flash the robot to make the changes permanent
